filebrowser_safe/views.py

```diff
@@ -98,7 +98,9 @@ def browse(request):
             raise ImproperlyConfigured(_("Error finding Upload-Folder. Maybe it does not exist?"))
         redirect_url = reverse("fb_browse") + query_helper(query, "", "dir")
         return HttpResponseRedirect(redirect_url)
-    abs_path = os.path.join(get_directory(), path)
+
+    base_dir = get_directory()
```

Comment:
`get_directory()` also requires a network call, and simply holding onto its result, here, made a significant difference.